test/rpc.test.js

```diff
@@ -68,26 +68,26 @@ describe('makeRpcRequest', function() {
     });
   });
 
-  it('should set the Content-Type header to application/json-rpc', function() {
+  it('should set the Content-Type header to application/json', function() {
     return makeRpcRequest(requestOptions)
     .then(function() {
-      expect(request.headers['content-type']).to.equal('application/json-rpc');
+      expect(request.headers['content-type']).to.equal('application/json');
     });
   });
 
   it('should set the Content-Length header to the byte length of the post body',
   function() {
     return makeRpcRequest(requestOptions)
     .then(function() {
-      var contentLength = Buffer.byteLength(requestBody, 'utf8');
+      var contentLength = Buffer.byteLength(JSON.stringify(requestBody), 'utf8');
       expect(request.headers['content-length']).to.equal(contentLength);
     });
   });
 
   it('should send a valid json-rpc 2.0 request object', function() {
     return makeRpcRequest(requestOptions)
     .then(function() {
-      var body = JSON.parse(requestBody);
+      var body = requestBody;
```

Comment:
Sorry to be a pain - but could we refer to `requestBody` in the following assertions instead of creating the renaming to `body`? Ditto for the next two changes too

Reply:
Oh yep, you're right.
I was too much focused on solving the issue that I forgot to read the code again.